lib/reek/smells/uncommunicative_parameter_name.rb

```diff
@@ -45,14 +40,15 @@ def self.default_config
       #
       # @return [Array<SmellWarning>]
       #
-      def examine_context(ctx)
-        context_expression = ctx.exp
-        context_expression.parameter_names.select do |name|
-          bad_name?(ctx, name) && ctx.uses_param?(name)
+      def examine_context(context)
+        expression = context.exp
+        expression.parameter_names.select do |name|
+          sanitized_name = sanitize name
+          !acceptable_name?(context: context, name: sanitized_name) && context.uses_param?(name)
```

Comment:
How about breaking out this complex expression into a method with an expressive name?